### DISCUSSION TITLE
Field Resolver

### DISCUSSION THREAD

**Discussion by edw19** [MichalLytek/type-graphql#941] [Q&A]
I have this Type created with type-graphql and according to documentation its posible call to the in @Fiel() insted of @FieldResolver(), I tried with both but i dont know what i'm doing wrong 

![image](https://user-images.githubusercontent.com/47750859/124287113-426a7000-db15-11eb-9ecf-c2d2b7c9803b.png)

in playground 
![image](https://user-images.githubusercontent.com/47750859/124287166-544c1300-db15-11eb-95cc-4bd4d9c80b3f.png)

in console, nothing
![image](https://user-images.githubusercontent.com/47750859/124287207-5e6e1180-db15-11eb-8c71-53bf8f4c2431.png)

also i tried this 
![image](https://user-images.githubusercontent.com/47750859/124287333-8198c100-db15-11eb-9a26-33fd8258091a.png)

**Comment by MichalLytek** [comment]
Show the `Vendor` type.

**Reply by edw19** [reply]
![image](https://user-images.githubusercontent.com/47750859/124299386-fa524a00-db22-11eb-8ccc-818d0afdbede.png)

**Comment by MichalLytek** [comment]
First of all, as shown in examples and explained in plenty of mongoose issues, you need to cast your documents into plain objects before returning from the query.

You can do that manually via `document.toObject()` or by using a middleware:
https://github.com/MichalLytek/type-graphql/blob/master/examples/typegoose/typegoose-middleware.ts

I don't know if this fixes your issue but it looks like the only invalid thing in the parts of the code you've shown.

**Reply by edw19** [reply]
I have the mongoose document transform settings, that doesn't seem to be the problem

**Reply by edw19** [reply]
I was working on solving this issue and came up with a solution that I don't know if it is correct. Could you check it please
first remove vendor field of type BUYS
![image](https://user-images.githubusercontent.com/47750859/124525132-ae720000-ddc3-11eb-8a98-445d14b64e05.png)

then add the buys type to @Resolver () of Vendors Resolver, no longer to BuysQueryResolver and finally add the @FieldResolver () with the name vendor in VendorsResolver with the type Vendor

![image](https://user-images.githubusercontent.com/47750859/124525272-496ada00-ddc4-11eb-9dc2-82ceb4ba3163.png)

![image](https://user-images.githubusercontent.com/47750859/124525299-61425e00-ddc4-11eb-9e49-5873978d9155.png)

:D :D
